src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/prometheus/common/log"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	vtls "github.com/vectorizedio/redpanda/src/go/rpk/pkg/tls"
+)
+
+const (
+	usersEndpoint = "/v1/security/users"
+	httpPrefix    = "http://"
+	httpsPrefix   = "https://"
+)
+
+type AdminAPI interface {
+	CreateUser(username, password string) error
+	DeleteUser(username string) error
+	ListUsers() ([]string, error)
+}
+
+type adminAPI struct {
+	url    string
+	client *http.Client
+}
+
+type newUser struct {
+	User      string `json:"username"`
+	Password  string `json:"password"`
+	Algorithm string `json:"algorithm"`
+}
+
+func NewAdminAPI(url string, tlsConf *config.TLS) (AdminAPI, error) {
+	var err error
+
+	// Go's http library requires that the URL have a protocol.
+	if !(strings.HasPrefix(url, httpPrefix) ||
+		strings.HasPrefix(url, httpsPrefix)) {
+
+		prefix := httpPrefix
+
+		if tlsConf != nil {
+			// If TLS will be enabled, use HTTPS as the protocol
+			prefix = httpsPrefix
+		}
+
+		url = fmt.Sprintf("%s%s", prefix, url)
+	}
+
+	var tlsConfig *tls.Config
+	if tlsConf != nil {
+		tlsConfig, err = vtls.BuildTLSConfig(
+			tlsConf.CertFile,
+			tlsConf.KeyFile,
+			tlsConf.TruststoreFile,
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+
+	client := &http.Client{Transport: tr}
+	return &adminAPI{url: url, client: client}, nil
+}
+
+func (a *adminAPI) CreateUser(username, password string) error {
+	if username == "" {
+		return errors.New("empty username")
+	}
+	if password == "" {
+		return errors.New("empty password")
+	}
+	u := newUser{
+		User:      username,
+		Password:  password,
+		Algorithm: sarama.SASLTypeSCRAMSHA256,
+	}
+	url := fmt.Sprintf("%s%s", a.url, usersEndpoint)
```

Comment:
are we sure the url ends with slash?

Reply:
Nope, but `usersEndpoint` starts with `/`: https://github.com/vectorizedio/redpanda/pull/1136/files/75c83d3a36ba963ea5fd826fbef69f3c134650bf#diff-fb0549b4e5b312851f56e8c829d12ac9d0bec63ab8ec2dce8dd9d59410ba9e91R32

It's a good point though, I should trim any trailing slashes. Thanks!

Reply:
Solved!